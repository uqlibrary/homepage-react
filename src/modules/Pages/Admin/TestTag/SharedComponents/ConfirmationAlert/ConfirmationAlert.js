import React from 'react';
import PropTypes from 'prop-types';
import Alert from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';

const ConfirmationAlert = ({ isOpen, message, type, closeAlert }) => {
    return (
        <Snackbar
            anchorOrigin={{
                vertical: 'top',
                horizontal: 'center',
            }}
            open={isOpen}
            autoHideDuration={6000}
            onClose={closeAlert}
            message={message}
        >
            <Alert onClose={closeAlert} severity={type}>
                {message}
            </Alert>
        </Snackbar>
    );
};

ConfirmationAlert.propTypes = {
    isOpen: PropTypes.bool,
    message: PropTypes.string,
    type: PropTypes.string,
    closeAlert: PropTypes.func,
};

ConfirmationAlert.defaultProps = {
    isOpen: false,
    message: '',
    type: 'info',
};

export default React.memo(ConfirmationAlert);
