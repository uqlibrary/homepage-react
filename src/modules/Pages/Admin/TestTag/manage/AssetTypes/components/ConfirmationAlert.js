import React from 'react';
import PropTypes from 'prop-types';

import Alert from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';
export const ConfirmationAlert = ({ isOpen, message, type, closeAlert, autoHide }) => {
    return (
        <Snackbar
            anchorOrigin={{
                vertical: 'top',
                horizontal: 'center',
            }}
            open={isOpen}
            autoHideDuration={!!autoHide ? 6000 : null}
            onClose={closeAlert}
            message={message}
        >
            <Alert onClose={closeAlert} severity={type}>
                {message}
            </Alert>
        </Snackbar>
    );
};

ConfirmationAlert.propTypes = {
    isOpen: PropTypes.bool,
    message: PropTypes.string,
    type: PropTypes.string,
    closeAlert: PropTypes.func,
    autoHide: PropTypes.bool,
};

ConfirmationAlert.defaultProps = {
    isOpen: false,
    message: '',
    type: 'info',
    autoHide: true,
};

export default React.memo(ConfirmationAlert);
