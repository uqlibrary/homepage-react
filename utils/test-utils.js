/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import preview from 'jest-preview';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
) => {
    return render(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const RenderWithRouter = ({
    children,
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
}) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

// eslint-disable-next-line react/prop-types
export const WithRedux = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>{children}</Provider>
);

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    WithRedux,
    WithReduxStore,
    RenderWithRouter,
    preview,
};
