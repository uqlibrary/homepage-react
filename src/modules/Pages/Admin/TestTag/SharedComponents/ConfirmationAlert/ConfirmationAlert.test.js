import React from 'react';
import ConfirmationAlert from './ConfirmationAlert';
import { render, act, fireEvent, waitFor } from 'test-utils';

/*
    isOpen: PropTypes.bool,
    message: PropTypes.string,
    type: PropTypes.string,
    closeAlert: PropTypes.func,
    */

function setup(testProps = {}, renderer = render) {
    const { isOpen = true, message = '', type = 'info', ...props } = testProps;
    return renderer(<ConfirmationAlert isOpen={isOpen} message={message} type={type} {...props} />);
}

describe('ConfirmationAlert', () => {
    it('renders empty component when closed', () => {
        const { queryByTestId } = setup({ isOpen: false });
        expect(queryByTestId('confirmation_alert-info')).not.toBeInTheDocument();
    });

    it('renders info component when open', () => {
        const { getByTestId } = setup({ message: 'Test message' });
        expect(getByTestId('confirmation_alert-info')).toBeInTheDocument();
        expect(getByTestId('confirmation_alert-info-alert')).toHaveTextContent('Test message');
    });
    it('renders success component when open', () => {
        const { getByTestId } = setup({ message: 'Test message', type: 'success' });
        expect(getByTestId('confirmation_alert-success')).toBeInTheDocument();
        expect(getByTestId('confirmation_alert-success-alert')).toHaveTextContent('Test message');
    });
    it('renders error component when open', () => {
        const { getByTestId } = setup({ message: 'Test message', type: 'error' });
        expect(getByTestId('confirmation_alert-error')).toBeInTheDocument();
        expect(getByTestId('confirmation_alert-error-alert')).toHaveTextContent('Test message');
    });
    it('fires close function automatically after provided timeout', async () => {
        const mockCloseFn = jest.fn();
        const { getByTestId } = setup({ message: 'Test message', closeAlert: mockCloseFn, autoHideDuration: 1000 });
        expect(getByTestId('confirmation_alert-info')).toBeInTheDocument();
        expect(getByTestId('confirmation_alert-info-alert')).toHaveTextContent('Test message');

        await waitFor(() => expect(mockCloseFn).toHaveBeenCalled(), { timeout: 3000 });
    });
    it('fires close function when close button clicked', async () => {
        const mockCloseFn = jest.fn();
        const { getByTestId, getByTitle } = setup({ message: 'Test message', closeAlert: mockCloseFn });
        expect(getByTestId('confirmation_alert-info')).toBeInTheDocument();
        expect(getByTestId('confirmation_alert-info-alert')).toHaveTextContent('Test message');

        act(() => {
            fireEvent.click(getByTitle('Close'));
        });
        expect(mockCloseFn).toHaveBeenCalled();
    });
});
