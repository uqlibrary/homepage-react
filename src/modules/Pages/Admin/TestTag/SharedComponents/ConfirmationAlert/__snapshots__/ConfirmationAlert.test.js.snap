// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationAlert default values render correctly 1`] = `<div />`;
